components/automate-ui/src/app/components/create-object-modal/create-object-modal.component.ts

```diff
@@ -52,37 +64,35 @@ export class CreateObjectModalComponent implements OnInit, OnDestroy, OnChanges
   }
 
   ngOnChanges(changes: SimpleChanges): void {
-    // update project dropdown if list changes
-    if (changes.assignableProjects) {
-      this.projects = {};
-      changes.assignableProjects.currentValue.forEach((proj: Project) =>
-        this.projects[proj.id] = { ...proj, checked: false });
-    }
-    // clear checked projects when opening
     if (changes.visible && (changes.visible.currentValue as boolean)) {
-      Object.values(this.projects).forEach(p => p.checked = false);
+
+      this.store.dispatch(new GetPolicies()); // refresh in case of updates
+
+      Object.values(this.projects).forEach(p => p.checked = false); // reset projects
       this.projectsUpdatedEvent.emit();
+
+      Object.values(this.policies).forEach(p => p.checked = false); // reset policies
+      this.policiesUpdatedEvent.emit();
+
       if (this.createProjectModal) {
-        this.updatePolicyCheckbox(true); // always set to checked upon opening
+        this.updatePolicyCheckbox(true); // default to checked upon opening
```

Comment:
trying to remember why we would need to check if it's a project modal is making me want to refactor this modal too 😅

Reply:
Good observation; I saw that too, but too much scope creep for here.